sdk/java/pulumi/src/main/java/com/pulumi/resources/StackOptions.java

```diff
@@ -1,33 +1,111 @@
 package com.pulumi.resources;
 
+import com.pulumi.core.internal.Copyable;
+
 import javax.annotation.Nullable;
 import javax.annotation.ParametersAreNonnullByDefault;
 import java.util.List;
 
+import static com.pulumi.resources.Resources.mergeNullableList;
+import static java.util.Objects.requireNonNull;
+
 /**
  * @see StackOptions is a bag of optional settings that control a stack's behavior.
  */
 @ParametersAreNonnullByDefault
-public class StackOptions {
+public class StackOptions implements Copyable<StackOptions> {
 
-    public static final StackOptions Empty = new StackOptions();
+    public static final StackOptions Empty = new Builder().build();
 
     private final List<ResourceTransformation> resourceTransformations;
 
-    public StackOptions() {
-        this.resourceTransformations = null;
-    }
-
-    public StackOptions(@Nullable List<ResourceTransformation> resourceTransformations) {
-        this.resourceTransformations = resourceTransformations;
+    /**
+     * Creates a {@link StackOptions} instance.
+     *
+     * @param resourceTransformations the resource transformations to use
+     * @see #builder()
+     */
+    public StackOptions(List<ResourceTransformation> resourceTransformations) {
+        this.resourceTransformations = requireNonNull(resourceTransformations);
     }
 
     /**
      * Optional list of transformations to apply to this stack's resources during construction.
      * The transformations are applied in order, and are applied after all the transformations of custom
      * and component resources in the stack.
+     *
+     * @return list of transformations to apply to children resources
      */
-    public List<ResourceTransformation> getResourceTransformations() {
-        return this.resourceTransformations == null ? List.of() : this.resourceTransformations;
+    public List<ResourceTransformation> resourceTransformations() {
+        return this.resourceTransformations;
+    }
+
+    public StackOptions copy() {
```

Comment:
You are right, this might be not needed anymore. The `copy` was carried over from C# where the first argument in merge methods is mutated after being copied.

In this builder I've actually changed that to crating a new StackOptions and making the field final, 
so I can rewrite to not use copy at all.

Thanks!